File: banchus-common/src/main/java/pe/nanamochi/banchus/redis/model/MultiplayerSlot.java
package pe.nanamochi.banchus.redis.model;

import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.domain.enums.SlotTeam;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MultiplayerSlot {
  private int slotId;
  private UUID sessionId;
  private int userId;
  private int status;
  private SlotTeam team;
  private int mods;
  private boolean loaded;
  private boolean skipped;
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/redis/entity/MultiplayerSlot.kt
package pe.nanamochi.banchus.redis.entity

import java.util.UUID
import pe.nanamochi.banchus.domain.enums.SlotTeam

data class MultiplayerSlot(
    var slotId: Int = 0,
    var sessionId: UUID? = null,
    var userId: Int = 0,
    var status: Byte = 0,
    var team: SlotTeam = SlotTeam.NEUTRAL,
    var mods: UInt = 0u,
    var isLoaded: Boolean = false,
    var isSkipped: Boolean = false,
)
